Improve the system reqs and installation instructions

radinamatic opened the pull request:
Addresses #15.

- [X] PEX is framed as Mac & other Linux distros installation method
- [X] PIP is still framed as an 'advanced' installation method with indication that it works on Mac, Windows, and Linux
- [X] Linux system requirements are still complex, but hopefully more accurate now
- [ ] The Debian support is *still* currently [only Python 3.4 and 3.5](https://github.com/learningequality/kolibri-installer-debian/issues/29). Do I add warning for it, or are we going to include support Python > 3.6? cc @aronasorman 

I Included @benjaoming's instructions for Raspberry from #14 so it can be merged together.

radinamatic committed:
First pass at de-confusion

radinamatic committed:
Rework the de-confusion

radinamatic committed:
Cleanup

indirectlylit commented on docs/install/_advanced.rst:
```diff
@@ -8,44 +8,21 @@ Advanced installation options
-  This section is pending instructions for creating a WI-FI hotspot. You can
-  install Kolibri on Raspbian following the above instructions for the PPA or
-  the ``pip install`` method.
+  This section is pending instructions for creating a WI-FI hotspot on Raspberry Pi. Follow the instructions above to install Kolibri on :ref:`Raspbian <rpi>`.
```
There two Raspberry Pi sections (here and under Linux). Should probably just delete this one?

The fact that we eventually want to have instructions on setting up a wifi hotspot is not helpful

indirectlylit commented on docs/install/_advanced.rst:
```diff
@@ -8,44 +8,21 @@ Advanced installation options
-  Django user, know that it is possible to use Kolibri with for instance
-  UWSGI+Nginx, using conventional methods.
-
+  This section is pending instructions for NGINX configuration. You can install Kolibri on Raspbian following the above instructions for the PPA or the ``pip install`` method. If you are an experienced system administrator or Django user, know that it is possible to use Kolibri with for instance UWSGI+Nginx, using conventional methods.
```
A few things still bother me about this section...

* it's not clear to me what Raspbian has to do with Ngninx?
* We should standardize on one capitalization, probably 'Nginx'
* having stuff like "This section is pending" to seems unhelpful and just adds noise

-----

![image](https://user-images.githubusercontent.com/2367265/38647964-c0fab848-3da3-11e8-8492-2358d320606b.png)

I'd recommend removing the entire 'advanced' section, and replacing it all with a single top-level "pip install" section.

indirectlylit commented on docs/install/_linux.rst:
```diff
@@ -37,18 +39,81 @@ The advantages of downloading a ``.deb`` file is the portability: you can copy t
+
+
+Troubleshooting
+^^^^^^^^^^^^^^^
```
Recommend making this an 'info' block so that it doesn't add noise to the TOC:

![image](https://user-images.githubusercontent.com/2367265/38648214-dd2fe9b0-3da4-11e8-8651-8b6260532fd0.png)

benjaoming replied:
There's an intention to create more sections from my side, possibly turning this into a "How-to" instead.

indirectlylit replied:
Might be best to break that into a separate file then so the top-level page stays simpler

indirectlylit commented:
> The Debian support is still currently only Python 3.4 and 3.5. Do I add warning for it, or are we going to include support Python > 3.6?

I think we should add a warning until we change what's supported. It's easy to update the docs so better to just make the accurate to the current state of the system?

As far as information structure for 'system requirements', here's my recommendation:

1. Move all client-side requirements to a different page, because they're not really  relevant here
   * Maybe more appropriate in the 'Access Kolibri' page, near the 'other devices' section?
   * Let's consolidate the 'Clients' and 'Supported Browsers' sections - as far as I can tell, they are the same thing

2. Instead of having a list of supported server operating systems at the top, maybe it would be easier to scatter the requirements across the various install methods? Then people first just look for their OS, and then see only the relevant instructions and compatibility?


Here's an example of what the final TOC might look like:


**Install Kolibri**

* Windows
  * Compatibility
  * Install
  * Uninstall
  * Upgrade
* Debian/Ubuntu
  * Compatibility
  * Install from PPA
  * Install from .deb file
  * Uninstall
  * Upgrade
* Raspbian
  * Compatibility
  * Install
  * Uninstall
  * Upgrade
* Other Linux & Mac
  * Compatibility
  * Install
  * Uninstall
  * Upgrade
* Python pip package
  * Compatibility
  * Install
  * Uninstall
  * Upgrade

indirectlylit commented:
Also, maybe we should just remove the Android instructions, and consolidate Mac with 'Other Linux'?

benjaoming commented:
As soon as this is released, we don't have to mention that .deb doesn't support Python 3.6: https://github.com/learningequality/kolibri-installer-debian/issues/29

If there are problems with Python 3.6, we can treat them as bugs and get them fixed :)

lyw07 commented on docs/install/_linux.rst:
```diff
@@ -37,18 +39,81 @@ The advantages of downloading a ``.deb`` file is the portability: you can copy t
+
+1. Running ``add-apt-repository`` as shown in the PPA instructions does not work. Instead, run::
+
+      echo "deb http://ppa.launchpad.net/learningequality/kolibri/ubuntu xenial main" > /etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list
```
@benjaoming I'm currently on a raspberry pi 3 with Raspbian Jessie. I run the command `echo "deb http://ppa.launchpad.net/learningequality/kolibri/ubuntu xenial main" > /etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list` but it errors out with `bash: /etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list: Permission denied`. I also try to run the command with `sudo`, but it doesn't work. I think the problem here is that the file `/etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list` has never been created. So I'm wondering before running this command, it would be better to create a file first? Do you have any suggestions? Thanks!

benjaoming replied:
Hi @lyw07 :)

Redirecting output to a file with `>` will create it when it doesn't exist. It's because `sudo` only binds to the first command, `sudo cmd > file` is the same as `(sudo cmd) > file`, `>` parses with a very low priority.

Anyways, this should work: `sudo su -c 'echo "deb http://ppa.launchpad.net/learningequality/kolibri/ubuntu xenial main" > /etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list'`

lyw07 replied:
ah thank you so much! that totally makes sense!

lyw07 replied:
@benjaoming I'm also wondering that instead of naming it a `raspberry pi` issue, would it be okay to name it a `debian jessie` issue? I tested on debian jessie days ago and found the same issue there: https://github.com/learningequality/kolibri-installer-debian/issues/30#issuecomment-381781511. My guess is that it's more likely a debian jessie issue?

benjaoming committed:
Thanks for noticing @lyw07

benjaoming commented:
Hi Lingyi!

Yes it's definitely a general Debian issue since the PPA doesn't contain
the various Debian dist names, so we have to manually specify "xenial"
or "trusty" in the apt source. Using `apt-add-repository` will
automatically use the current dist name, which isn't sufficient when it
isn't Ubuntu.

Best,
Ben


On 04/18/2018 05:58 PM, Lingyi Wang wrote:
>
> *@lyw07* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In docs/install/_linux.rst
> <https://github.com/learningequality/kolibri-docs/pull/16#discussion_r182480450>:
>
> >  
>  When you use the PPA installation method, upgrades to newer versions will be automatic, provided there is internet access available.
>  
>  To upgrade Kolibri on a Debian device without internet access, bring the updated ``.deb`` file and follow the same steps as in :ref:`lin_deb`.
> +
> +
> +.. _rpi:
> +
> +Raspberry Pi (Raspbian)
> +-----------------------
> +
> +Kolibri currently doesn't work out of the box for Raspbian Jessie. We are refining our distribution to work out of the box, but you need to follow these two steps in order to install it (tested on Kolibri 0.9):
> +
> +1. Running ``add-apt-repository`` as shown in the PPA instructions does not work. Instead, run::
> +
> +      echo "deb http://ppa.launchpad.net/learningequality/kolibri/ubuntu xenial main" > /etc/apt/sources.list.d/learningequality-ubuntu-kolibri-xenial.list
>
> @benjaoming <https://github.com/benjaoming> I'm also wondering that
> instead of naming it a |raspberry pi| issue, would it be okay to name
> it a |debian jessie| issue? I tested on debian jessie days ago and
> found the same issue there:
> learningequality/kolibri-installer-debian#30 (comment)
> <https://github.com/learningequality/kolibri-installer-debian/issues/30#issuecomment-381781511>.
> My guess is that it's more likely a debian jessie issue?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/learningequality/kolibri-docs/pull/16#discussion_r182480450>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAW3VAgjCUhVLDiw8rO4CL_vFnSCh_ncks5tp2KRgaJpZM4TPfRt>.
>

lyw07 commented:
@benjaoming Thank you for the explanation! I agree that `apt-add-repository` needs to be replaced by manually specifying dist names. But if I remember correctly, `apt-add-repository` works on Debian Stretch as it gets `Ubuntu Bionic`. Have you had problem running `apt-add-repository` on Debian Stretch as well? Thank you!

radinamatic committed:
Compatibility per OS

radinamatic committed:
Merge branch 'install-deconfusion' of github.com:radinamatic/kolibri-docs into install-deconfusion

indirectlylit commented:
awesome, thanks! what do you think of this organization, @radinamatic?

Btw, travis is [failing](https://travis-ci.org/learningequality/kolibri-docs/builds/370335859?utm_source=github_status&utm_medium=notification):

```
Warning, treated as error:
/home/travis/build/learningequality/kolibri-docs/docs/access.rst:78:Title level inconsistent:
```

radinamatic committed:
Improved access on Linux & MacOS, trimmed system reqs

radinamatic committed:
Separate Raspbian

radinamatic commented:
@indirectlylit I tried to include all your suggestions:
- OS/browser reqs are now indicated per each platform, client ones moved to the `Access page`
- Eliminated "Advanced installation" and added `pip install` as top level item
- Added the new `How to...` section, invisible for now, for when we have the RPi-wifi and Nginx sections ready

@lyw07 Can I get your eyes on the `Raspbian` section? Haven't done these steps myself, and your review will be the final word! :slightly_smiling_face:

lyw07 commented on docs/install/_rpi.rst:
```diff
@@ -0,0 +1,55 @@
+
+2. Kolibri does not start after installation. This is because ``python3-cffi`` is outdated on Raspbian. Upgrade it like this::
+
+      sudo apt install libffi-dev
```
Not sure why we need `libffi-dev`... I tried the installation steps without this step, and kolibri works fine.

radinamatic replied:
That was verbatim the original RPi installation instructions by @benjaoming from #14. If you are positive it's not necessary, I can delete that line.

lyw07 replied:
I see. Thanks! Then maybe we can leave it here for now before Ben comes back to see if kolibri works on his machine without this step? The result could differ by different machines.

radinamatic replied:
Sounds good! :+1: 

@indirectlylit Ready for the final review! :wink:

indirectlylit commented:
awesome, thanks @radinamatic!

indirectlylit commented on docs/access.rst:
```diff
@@ -7,7 +7,7 @@ Access Kolibri
 ===========================
 
-To start **Kolibri** on Windows, just double-click the desktop shortcut. You will see the notification message *Kolibri is starting, please wait...*.
+To start **Kolibri** server on Windows, just double-click the desktop shortcut. You will see the notification message *Kolibri is starting, please wait...*.
```
> To start Kolibri server on Windows

Maybe "To start **the** Kolibri server on Windows"?

indirectlylit commented on docs/access.rst:
```diff
@@ -53,40 +61,49 @@ Starting Kolibri on Linux
+* MacOS 10.6+ with Safari, Chrome and Firefox
+* Linux, any browser
+* Android 4.2+, Chrome and Firefox
+* iOS ``XXX``, Chrome and Firefox supported, **Safari not supported**
```
maybe remove ``XXX``

radinamatic replied:
No version number on iOS?

indirectlylit replied:
I don't think we know...

indirectlylit commented on docs/install.rst:
```diff
@@ -3,9 +3,9 @@
+See the `Kolibri changelog here <https://learningequality.org/r/kolibri-changes>`_. Documentation for `previous Kolibri versions are available here <https://readthedocs.org/projects/kolibri/>`_.
 
-To install **Kolibri**, check the system requirements first and then follow the procedure for the operating system on your device.
+To install **Kolibri** on device that will act as server, check the hardware requirements first and then follow the procedure for the operating system on your device.
```
>  on device that will act as server

maybe "on **a** device that will act as **a** server"?

indirectlylit commented on docs/install/_rpi.rst:
```diff
@@ -0,0 +1,55 @@
+Compatibility
+-------------
+
+Kolibri currently doesn't work out of the box for Raspbian Jessie. We are refining our distribution to work out of the box, but you need to follow the below steps in order to install it (tested on Kolibri 0.9).
```
> but you need to follow

"but you can follow"

indirectlylit commented on docs/install/_windows.rst:
```diff
@@ -3,7 +3,12 @@
+-------------
+
+* Supported: Windows 7, 8.1 and 10, with IE 11+, Chrome or Firefox
+* **Not supported: Windows XP** cannot be used to install Kolibri server, but could potentially work as a client device if the browsers are `as up-to-date as possible <https://support.mozilla.org/en-US/questions/1173904>`_.
```
The bolding on

```
**Not supported: Windows XP** cannot be used
```

is a bit strange. Maybe

```
**Not supported:** Windows XP cannot be used
```

indirectlylit commented on docs/manage/_how_to.rst:
```diff
@@ -0,0 +1,20 @@
+---------------
+
+.. note::
+  If you are an experienced system administrator or Django user, know that it is possible to use Kolibri with for instance UWSGI+Nginx, using conventional methods.
```
> If you are an experienced system administrator or Django user, know that it is possible to use Kolibri with for instance UWSGI+Nginx, using conventional methods.

Maybe

> If you are an experienced system administrator or Django user, it is possible to deploy Kolibri using conventional methods - for instance with uWSGI and Nginx.

indirectlylit reviewed (approved):
Looks great! just a few minor tweaks

radinamatic committed:
Last suggested edits

radinamatic commented:
@indirectlylit Done, done & done! :slightly_smiling_face:

benjaoming commented:
Hey, I need the RPi instructions to be live, but can see that the final review from @indirectlylit  is marked done - hoping that it's okay to merge :)

indirectlylit commented:
great upgrade - thanks all
